### PR TITLE
updating the bundle version resolution to pick the maximum version allowed by the version range

### PR DESCRIPTION
**Pull request by soninaren**

### CODE REVIEW

File: src/WebJobs.Script/ExtensionBundle/ExtensionBundleManager.cs [hunk 1]
```diff
@@ -190,7 +190,16 @@ private static string FindBestVersionMatch(VersionRange versionRange, IEnumerabl
                 return version;
             }).Where(v => v != null);
 
-            return versionRange.FindBestMatch(bundleVersions)?.ToString();
+            var sortedBundleVersions = bundleVersions.OrderByDescending(version => version.Version);
+            for (int i = 0; i < sortedBundleVersions.Count(); i++)
```
Comment (pragnagopa) [inline src/WebJobs.Script/ExtensionBundle/ExtensionBundleManager.cs]: As we are using nuget version semantics, it is better to not change how a version range is interpreted. Why not just enforce [2.*, 3.0.0) format if users need to get the latest version available?

Comment (fabiocav) [reply src/WebJobs.Script/ExtensionBundle/ExtensionBundleManager.cs]: We're using nuget APIs and applying the version range, which still applies, but the version selection behavior is not tied to them (this is not documented or advertised as being NuGet version selection).

Comment (fabiocav) [inline src/WebJobs.Script/ExtensionBundle/ExtensionBundleManager.cs]: Can we not change this logic to use `FirstOrDefault` ?

Comment (pragnagopa) [reply src/WebJobs.Script/ExtensionBundle/ExtensionBundleManager.cs]: Ok. We do have to document what version ranges mean similar to https://docs.microsoft.com/en-us/nuget/reference/package-versioning#version-ranges-and-wildcards

**Pull request merged**